folksonomy/api.py

```diff
@@ -81,11 +81,7 @@ async def app_lifespan(app: FastAPI):
     # So, for everything to work correctly, it's better to specify explicitly the allowed origins."
     # => Workarround: use allow_origin_regex
     # Source: https://github.com/tiangolo/fastapi/issues/133#issuecomment-646985050
-    allow_origins=[
-        "http://localhost:8000",
-        "http://world.openfoodfacts.localhost",
-    ],
-    allow_origin_regex="https?://.*",
+    allow_origin_regex=r"^https://.*|^http://(localhost|.*\.localhost)(:\d+)?$",
```

Comment:
What's your reasoning for doing `s://` instead of `s?://`?

Also, the second part could probably be condensed to something like
```suggestion
    allow_origin_regex=r"^https://.*|^http://(.*\.)?localhost(:\d+)?$",
```

Or the whole thing to something like

```suggestion
    allow_origin_regex=r"^http(s://.*|://(.*\.)?localhost(:\d+)?)$",
```

Though that’s a bit wonky looking. Did you try something like

```suggestion
    allow_origin_regex=r"^https?://(.*|(.*\.)?localhost)(:\d+)?$",
```

? Maybe that won’t work, but if it does, I think that’s my favourite variation. Actual maintainers of this might have other thoughts though, so don’t put too much weight on this one comment. :)